### PR TITLE
add serviceMappingListener for application service discovery

### PR DESCRIPTION
**Pull request by bobtthp**
issue: https://github.com/apache/dubbo-go/issues/2167

support add  service mapping listen for service change

### CODE REVIEW

**Comment by justxuewei** [comment]
It looks like this pull request breaks the ci tests, please fix it.

**Review by justxuewei — APPROVED** [review]
Lgtm, thanks!

File: metadata/report/nacos/report.go [hunk 1]
```diff
@@ -209,6 +210,34 @@ func (n *nacosMetadataReport) getConfig(param vo.ConfigParam) (string, error) {
...
+func (n *nacosMetadataReport) removeServiceMappingListener(key string, group string) error {
+	return n.client.Client().CancelListenConfig(vo.ConfigParam{
+		DataId: key,
+		Group:  group,
+	})
+}
+
 // RegisterServiceAppMapping map the specified Dubbo service interface to current Dubbo app name
 func (n *nacosMetadataReport) RegisterServiceAppMapping(key string, group string, value string) error {
 	oldVal, err := n.getConfig(vo.ConfigParam{
```
Comment (mark4z) [inline metadata/report/nacos/report.go:243]: OldVal is a list of service names. If strings.Contains(oldVal, newServiceName), then the SNP registration will be skipped, e.g. if oldVal = dubbo.io-2 and newServiceName = dubbo.io, then newServiceName will be skipped.

**Review by mark4z — APPROVED** [review]

**Comment by mark4z** [comment]
see https://github.com/apache/dubbo/pull/11643/files

**Comment by bobtthp** [comment]
> see https://github.com/apache/dubbo/pull/11643/files

i will create a new issue and pr to resolve this prom

**Comment by bobtthp** [comment]
> see https://github.com/apache/dubbo/pull/11643/files

done

**Comment by mark4z** [comment]
lgtm

**Review by justxuewei — APPROVED** [review]

**Pull request merged**